src/record/record.js

```diff
@@ -449,7 +446,16 @@ Record.prototype._applyUpdate = function( message ) {
  */
 Record.prototype._onRead = function( message ) {
 	this.version = parseInt( message.data[ 1 ], 10 );
-	this._applyChange( jsonPath.set( this._$data, undefined, JSON.parse( message.data[ 2 ] ) ) );
+
+	this._$data = JSON.parse( message.data[ 2 ] );
```

Comment:
Your last example is incorrect:

1) Subscribe to path name
**2) name triggered with undefined** Nope
3) Get read -> {}
4) name triggered with undefined